packages/@react-aria/slider/src/useSliderThumb.ts

```diff
@@ -77,44 +77,77 @@ export function useSliderThumb(
   stateRef.current = state;
   let reverseX = direction === 'rtl';
   let currentPosition = useRef<number>(null);
+
+  let {keyboardProps} = useKeyboard({
+    onKeyDown(e) {
+      let {
+        getThumbMaxValue,
+        getThumbMinValue,
+        pageSize
+      } = stateRef.current;
+      // these are the cases that useMove or useSlider don't handle
+      if (!/^(PageUp|PageDown|Home|End)$/.test(e.key)) {
+        e.continuePropagation();
+        return;
+      }
+      // same handling as useMove, stopPropagation to prevent useSlider from handling the event as well.
+      e.preventDefault();
+      // remember to set this so that onChangeEnd is fired
+      state.setThumbDragging(index, true);
+      switch (e.key) {
+        case 'PageUp':
+          stateRef.current.incrementThumb(index, pageSize);
+          break;
+        case 'PageDown':
+          stateRef.current.decrementThumb(index, pageSize);
+          break;
+        case 'Home':
+          state.setThumbValue(index, getThumbMinValue(index));
+          break;
+        case 'End':
+          state.setThumbValue(index, getThumbMaxValue(index));
+          break;
+      }
+      state.setThumbDragging(index, false);
+    }
+  });
+
   let {moveProps} = useMove({
-    onMoveStart({pointerType}) {
+    onMoveStart() {
       currentPosition.current = null;
       // Don't start dragging for keyboard events unless the value has changed.
-      if (pointerType !== 'keyboard') {
-        stateRef.current.setThumbDragging(index, true);
-      }
+      stateRef.current.setThumbDragging(index, true);
     },
-    onMove({deltaX, deltaY, pointerType}) {
+    onMove({deltaX, deltaY, pointerType, shiftKey}) {
       const {
-        getThumbPercent, 
-        getThumbMaxValue, 
-        getThumbMinValue, 
-        getThumbValue, 
-        setThumbDragging, 
-        setThumbPercent, 
-        setThumbValue, 
-        step
+        getThumbPercent,
+        setThumbPercent,
+        step,
+        pageSize
       } = stateRef.current;
       let size = isVertical ? trackRef.current.offsetHeight : trackRef.current.offsetWidth;
 
       if (currentPosition.current == null) {
         currentPosition.current = getThumbPercent(index) * size;
       }
       if (pointerType === 'keyboard') {
-        const currentValue = getThumbValue(index);
-        // (invert left/right according to language direction) + (according to vertical)
-        const delta = ((reverseX ? -deltaX : deltaX) + (isVertical ? -deltaY : -deltaY)) * step;
-        if (delta > 0 && currentValue === getThumbMaxValue(index)) {
-          return;
-        }
-        if (delta < 0 && currentValue === getThumbMinValue(index)) {
-          return;
+        if (deltaX > 0) {
+          if (direction === 'rtl') {
```

Comment:
We have `reverseX` declared above at https://github.com/adobe/react-spectrum/pull/2819/files#diff-9eebdd2756162e835cc5a37501ce3ccc3decdfd5d1dee4345d2e5f5a06c38703R78